packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts

```diff
@@ -1,17 +1,27 @@
+import Web3 from "web3";
+const { toBN } = Web3.utils;
+
 import { ZERO_ADDRESS } from "@uma/common";
 import { getAbi } from "@uma/contracts-node";
+import { Deposit } from "./InsuredBridgeL2Client";
+
 import type { BridgeAdminWeb3, BridgePoolWeb3 } from "@uma/contracts-node";
-import Web3 from "web3";
+import type { BN } from "@uma/common";
 import type { Logger } from "winston";
-import { Deposit } from "./InsuredBridgeL2Client";
 
-enum relayState {
+enum RelayState {
   Pending,
   SpedUp,
   Disputed,
   Finalized,
 }
 
+export enum RelayAbility {
```

Comment:
This is kind of a weird name but I get what you're going for. You want to enumerate whether a Relay can be submitted, and what type. Why can't you just re-use the following `RelayState` types:

```
Any --> Pending || (some state before Pending like Uninitialized)
SpeedUpOnly --> Pending
None --> SpedUp || Disputed || Finalized
```

You'd only need to add an Uninitialized state

Reply:
Or another idea, since I also don't like the idea of mixing _contract_ state and off-chain client state, rename this enum to `ClientRelayState {Uninitialized, Pending, Finalized}` to distinguish between on-chain and off-chain state. We could moreover rename `RelayState` to `ContractRelayState`

Reply:
ye, this is something I went back and forth a few times on and is something that is not immediately clear from the review. as you pointed out, we already have `RelayState` which contains pretty much the states you discussed:
```ts
enum RelayState {
  Pending,
  SpedUp,
  Disputed,
  Finalized,
}
```

my thinking with the `RelayAbility` structure is that there is a state _before_ `RelayState.Pending` when the deposit has been sent on L2 but nothing has been brought onto L1. this is encapsulated by `RelayAbility.Any` meaning that the state is pre-relay.

With that said, `ClientRelayState {Uninitialized, Pending, Finalized}` seems pretty reasonable and fits the same framing that I require. I've refactored accordingly :)